ethcore/src/miner/transaction_queue.rs

```diff
@@ -794,16 +813,45 @@ impl TransactionQueue {
 	}
 
 	/// Checks the current nonce for all transactions' senders in the queue and removes the old transactions.
-	pub fn remove_old<F>(&mut self, fetch_nonce: F) where
-		F: Fn(&Address) -> U256,
+	pub fn remove_old<F>(&mut self, fetch_account: &F, current_time: InsertionTime) where
+		F: Fn(&Address) -> AccountDetails,
 	{
 		let senders = self.current.by_address.keys()
 			.chain(self.future.by_address.keys())
-			.cloned()
-			.collect::<HashSet<_>>();
+			.map(|sender| (*sender, fetch_account(sender)))
+			.collect::<HashMap<_, _>>();
 
-		for sender in senders {
-			self.remove_all(sender, fetch_nonce(&sender));
+		for (sender, details) in senders.iter() {
+			self.remove_all(*sender, details.nonce);
```

Comment:
`remove_all` is a rather misleading name. there is very specific logic here that the name should imply. suggest `remove_dead` or even just `cull`.